make compatible with TYPO3 12

Kanti opened the pull request:
Tested in TYPO3 11 and 12

Kanti committed:
[BUGFIX] Make TYPO3 12 compatible

sascha-egerer commented on Configuration/Services.yaml:
```diff
@@ -37,3 +47,10 @@ services:
+        tags:
+            -
+                name: 'event.listener'
+                identifier: 'helhum/t-y-p-o3/crontab/before-module-creation-event-listener'
```
```suggestion
                identifier: 'helhum/crontab/before-module-creation-event-listener'
```

`/t-y-p-o3/`?

Kanti replied:
it was auto generated by b13/make

Kanti committed:
Update Configuration/Services.yaml

Co-authored-by: Sascha Egerer <sascha.egerer@gmail.com>

linawolf commented:
@helhum what is the plan here?

linawolf commented:
According to me tests this solution works with scheduler tasks but not with console commands as there will be a type error in the CommandExecutor. @Kanti have you tested your code in TYPO3 v12 with a console command?

Kanti commented:
We are running this in Production for about 6 Months without Problems. (with Console Commands)
What is the error you are getting? with what PHP Version?

linawolf commented:
In the CommandExecutor, line 37 an array is passed to ArgvInput which expects an InputInterface or null

Tim-Obert committed:
Fix Input for run in CommandExecutor

There is an exception if the ArgvInput is used from HelhumTypo3Console, because it expects an InputInterface and no array

Kanti committed:
Merge pull request #1 from Tim-Obert/v12

Kanti committed:
[TASK] allow php 8.3

devtroll commented:
@linawolf @Kanti Merge Status to Main?

Kanti commented:
I see this extension has 2 features.

-  It allowed to commit the defined scheduler tasks.
- and it tracks the processid and marked task as not running If they had a problem 

One of these features can possible done with events in the scheduler. (Processid)

The other one can be done with an import into the scheduler tasks table.

I think an extension that dose both of these things should be a good solution without much implementation. And using the existing infrastructure from the Core/Scheduler.

The feature that all task are executed in separate process could maybe brought into the core or be done with an xclass. I think that should not be a priority.

devtroll commented:
@Kanti The Main Problem of TYPO3 Scheduler is the Sharing between multiple Instances. That is Not possible

helhum commented:
I was not able to push additional changes on top of this, so I added those commits to #18 and will close this one now

helhum commented:
Thanks a ton @Kanti for the preparation! Thanks everybody for your patience!
